### PR TITLE
fix: don't hide video if it's going to play next

### PR DESCRIPTION
**Pull request by actionless**

### CODE REVIEW

**Comment by EvHaus** [comment]
I'm not sure I understand the intention for this change. On which pages does this happen?

**Comment by actionless** [comment]
on `/watch?` pages

**Comment by actionless** [comment]
to reproduce the problem from the second commit, set dev const to enable logging

File: main.js [hunk 1]
```diff
@@ -222,13 +222,16 @@ html[dark] .YT-HWV-BUTTON {
 				watchedItem = item.closest('.ytd-grid-renderer');
 			} else if (section === 'playlist') {
 				watchedItem = item.closest('ytd-playlist-video-renderer');
+			} else if (section === 'watch') {
+				watchedItem = item.closest('ytd-compact-video-renderer');
+				// don't hide video if it's going to play next:
+				if (watchedItem.closest('ytd-compact-autoplay-renderer')) watchedItem = null;
```
Comment (EvHaus) [inline main.js]: On the watch page this line is failing for me with:

```
userscript.html?id=9152cc27-b376-4b61-adbd-89964ca68e0a:234 Uncaught TypeError: Cannot read property 'closest' of null
    at eval (userscript.html?id=9152cc27-b376-4b61-adbd-89964ca68e0a:234)
    at Array.forEach (<anonymous>)
    at updateClassOnWatchedItems (userscript.html?id=9152cc27-b376-4b61-adbd-89964ca68e0a:203)
    at eval (userscript.html?id=9152cc27-b376-4b61-adbd-89964ca68e0a:322)
    at Window.later (userscript.html?id=9152cc27-b376-4b61-adbd-89964ca68e0a:135)
    at <anonymous>:3:75
    at c.<computed> (eval at exec_fn ((index):1), <anonymous>:55:204)
```

You can reproduce on this page: https://www.youtube.com/watch?v=H8IaDYrv-3E&list=PL9Zb2rdpvfzozn0Ar1UFbxOg0tsgdHQeT&index=39&t=0s

File: main.js [hunk 2]
```diff
@@ -299,7 +302,16 @@ html[dark] .YT-HWV-BUTTON {
 		button.setAttribute('title', `Toggle Watched Videos (currently "${state}" for "${section}" section)`);
 	};
 
-	const run = debounce(() => {
+	const run = debounce((mutations) => {
+		if (
```
Comment (EvHaus) [inline main.js]: Good catch with this. I see the issue now! 👍

File: main.js (hunk 1 above)
Comment (actionless) [reply main.js]: done

**Comment by EvHaus** [comment]
LGTM. Thanks for another contribution!

**Pull request merged**